File: src/main/webapp/res/js/user/recommendService.js
app.service("recommendService", function ($http) {
    this.edit = function (_data) {
        var req = {
            method: 'POST',
            url: context + '/user/recommend/edit.rest',
            headers: {
                'Content-Type': 'application/json'
            },
            data: _data
        };
        return $http(req);
    }
    this.getRecommend = function (_id) {
        var req = {
            method: 'POST',
            url: context + '/user/recommend/findById.rest?id=' + _id,
            headers: {
                'Content-Type': 'application/json'
            }
        };
        return $http(req);
    }

    this.initList = function (_index, _size) {
        var param = "";
        if (_index != undefined) {
            param += "index=" + _index;
        }
        if (_size != undefined) {
            param += "&size=" + _size;
        }
        var req = {
            method: 'POST',
            url: context + '/user/recommend/list.rest?' + param,
            headers: {
                'Content-Type': 'application/json'
            }
        };
        return $http(req);
    }
    this.delete = function (_id) {
        var req = {
            method: 'POST',
            url: context + '/user/recommend/delete.rest?id=' + _id,
            headers: {
                'Content-Type': 'application/json',
            }
        }
        return $http(req);
    }
})

File: src/main/java/com/hu/dg/service/RecommendService.java
package com.hu.dg.service;

import com.hu.dg.domain.Recommend;
import com.hu.dg.domain.User;

import java.util.List;
import java.util.Map;

/**
 * 类的功能，目的，描述等写在此处
 *
 * @author 胡玉波
 * @version 1.0
 * @(#) RecommendService.java 2016/06/29 19:56
 */
public interface RecommendService {

    void edit(Recommend recommend,User user);

    List<Recommend> list(Map<String,Object> params);

    void delete(int id);

    int getCount(Map<String,Object> params);

    Recommend findById(int id);
}